### DISCUSSION TITLE
Print total number of issues, even if maxIssues is not reached, exit code 0

### DISCUSSION THREAD

**Discussion by sjjctl** [detekt/detekt#5957] [Q&A] [answered]
Your feature request may already be reported!
Please search in the [issues](https://github.com/detekt/detekt/issues) before creating a new one.

## Current Behavior

Currently, if the `maxIssues` threshold isn't reached, the process exits cleanly and doesn't output the same `"Analysis failed with n issues"` message indicating the total number

## Context

We check the total number against the allowed number in our CI pipelines, to make sure if a new lower bound is reached that we configure out `detekt.yml` to make that the new `maxIssues`.

But it's not easy to do this without either running the process twice, or piping `stderr` to a file and parsing the error count that way.

**Comment by BraisGabin** [comment, marked as answer]
Why don't you use a baseline instead? I have a baseline with all the current errors and set the allows errors to 0. The baseline suppresses all those known errors but doesn't allow new ones.

Then I also have a nightly that regenerates that baseline. Because we don't allow to merge anything with new errors that task only removes the fixed things from the baseline to avoid regressions.

**Comment by schalkms** [comment]
I converted this to a discussion.
Also, I agree with Brais here.

**Comment by schalkms** [comment]
Further discussion is welcome. 🙂

**Comment by sjjctl** [comment]
How often does a baseline need updating? What if a function is renamed, etc. I think this should be an option, not a requirement.

We should consider the pros and cons of outputting an error/warning count even when the check passes (this is how `eslint` works). We can at least add a CLI option for it, if not making it the default behavior. Thoughts?

I ended up solving this a different way (which has its own limitations), by ignoring unmodified files and allowing zero issues (so either resolve or suppress any new ones). For small PRs this also dramatically speeds up the linting process.

See the below `Makefile` syntax,

```makefile
DETEKT_END_FILE ?= ~/.local/lib/detekt-cli-1.22.0-all.jar

REMOTE_HEAD ?= origin/master
CHANGED_FILES_KT ?= $(shell git diff $(REMOTE_HEAD) --name-only --diff-filter=MACRU \*.kt \*.kts)

.PHONY: lint
lint:	## run Detekt on changed .kt/.kts files
	# Lint any changed .kt or .kts files
	if [ "$(CHANGED_FILES_KT)" ]; then \
	  java -jar $(DETEKT_END_FILE) --parallel --max-issues 0 -i $(shell echo $(CHANGED_FILES_KT) | tr ' ' ','); \
	fi
```

**Reply by BraisGabin** [reply]
For sure, everything could be implemented but we should focus on common scenarios. Baseline fixes your issue. Your exposed solution fixes your issue. And, if you want, you can implement your own console report and report whatever you want.
